### DISCUSSION TITLE
Is TSX support still planned for 2024 Q1?

### DISCUSSION THREAD

**Discussion by MSDarwish2000** [material-components/material-web#5550] [Q&A] [answered]
The [roadmap](https://github.com/material-components/material-web/blob/main/docs/roadmap.md) mentions that TSX support is planned for 2024 Q1 which ends in 3 days with no mentions of this feature despite recent release of v1.4.0. May I ask if it was rescheduled or is it expected to be released within a short period?

I'm asking as I'm going to start a new small project using React+NextJS and seeing this library as a good resource to use but I prefer avoiding workarounds.

**Comment by asyncliz** [comment, marked as answer]
TSX support is delayed due to Material Design shifting resources around. I'm hoping we can update our roadmap within the next couple of weeks.

However, React support is fairly simple to add yourself. https://github.com/material-components/material-web/discussions/5186 gives some guidance how to use MWC in React right now. The "official" support from us would just be using `@lit-labs/react` to set up the boilerplate for y'all.

@anthonyleung-dev [did some work](https://github.com/material-components/material-web/discussions/5544#discussioncomment-8933339) with their own React support. https://github.com/anthonyleung-dev/react-material-web shows their effort that can be referenced.

**Reply by MSDarwish2000** [reply]
Thank you for your response

**Reply by TKDev7** [reply]
Liz can you explain what you mean by “ Material Design shifting resources around”?

**Comment by JosefJezek** [comment]
please update roadmap
